### PR TITLE
Added Sales summary interactive query

### PR DESCRIPTION
**Pull request by pedrooste**

### CODE REVIEW

File: src/main/java/com/mercuryCyclists/businessIntelligence/service/SaleStreamProcessing.java [hunk 1]
```diff
@@ -35,7 +36,7 @@ public Consumer<KStream<?, SaleEvent>> process() {
                 String productName = v.getProductName();
                 String quantity = v.getQuantity().toString();
                 String price = v.getPrice().toString();
-                String newKey = productName + "_" + quantity + "_" + price;
+                String newKey = UUID.randomUUID().toString();
```
Comment (pedrooste) [inline src/main/java/com/mercuryCyclists/businessIntelligence/service/SaleStreamProcessing.java:39]: This is the sketchy part, we dont have anything thats unique

Comment (Jack-Is-2048) [reply src/main/java/com/mercuryCyclists/businessIntelligence/service/SaleStreamProcessing.java:39]: Well the key is never used only the value, so I wouldn't stress it

File: src/main/java/com/mercuryCyclists/businessIntelligence/controller/SaleQueryController.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    @Autowired
+    public SaleQueryController(SaleInteractiveQuery saleInteractiveQuery) {
+        this.saleInteractiveQuery = saleInteractiveQuery;
+    }
+
+    @GetMapping("/summary")
+    public Collection<SaleSummary> getSaleSummary() {
+        return saleInteractiveQuery.getSaleSummary();
+    }
```
Comment (Jack-Is-2048) [inline src/main/java/com/mercuryCyclists/businessIntelligence/controller/SaleQueryController.java:28]: What if I want to see a summary for a specific product?

File: src/main/java/com/mercuryCyclists/businessIntelligence/service/SaleStreamProcessing.java (hunk 1 above)
Comment (pedrooste) [reply src/main/java/com/mercuryCyclists/businessIntelligence/service/SaleStreamProcessing.java:39]: needs to be unique, but yea

**Review by Jack-Is-2048 — APPROVED** [review]

**Review by Jack-Is-2048 — APPROVED** [review]

**Pull request merged**